### ISSUE TITLE
multiple readings

### ISSUE THREAD

**Issue by mickrose555** [newenclave/AtomAppGN#2]
for your info: iOS atom app allows to read 4x Atomfast at the same time. BLE signal is just displayed in cps, live view. then you press one and you get the graph...
Any chance to display more than 1 reading?

**Comment by newenclave** [comment]
I'm not ready to answer it right now :) 
What I understood from the SDK reading, the Garmin devices are tiring to be "power safe" devices. I'm not sure if the device will allow me to pair with more then one device. But, again, I've not dived into so deep yet.

**Comment by mickrose555** [comment]
i saw the youtube video.
1) can you just explain if you believe your beta version can keep a connection for 20min?
2) do you see a difference in reaction time between an Android or the watch?
thanks

**Comment by newenclave** [comment]
> can you just explain if you believe your beta version can keep a connection for 20min?

What do you mean? The application can be running as long as you want. Or as long as the battery allows.  Ok. I'll try to run it and keep it on for 20-30 minutes.

> do you see a difference in reaction time between an Android or the watch?

The Atom Fast device notify about any changes every 2 seconds. So the application updates values when it receives changes.

**Comment by mickrose555** [comment]
> > can you just explain if you believe your beta version can keep a connection for 20min?
> 
> 
> 
> What do you mean? The application can be running as long as you want. Or as long as the battery allows.  Ok. I'll try to run it and keep it on for 20-30 minutes.
> 
exactly! for your info, some iOS versions with Atom would loose connection after 5 or 10min, very annoying when it drops like that. 
> 
> 
> > do you see a difference in reaction time between an Android or the watch?
> 
> 
> 
> The Atom Fast device notify about any changes every 2 seconds. So the application updates values when it receives changes. 
good. as long as it is similar to android app, we're good and BLE update works.

**Comment by newenclave** [comment]
Ok. I've been testing it for an hour. No issues were found. I tried to disconnect it (moved it outside, wait for couple of minutes and brought it back). The device reconnected it with no help from my side. Looks quite stable and tough. But as I can said, the application is soo simple and basic for now :)

**Comment by mickrose555** [comment]
2 more questions:
i have up to 6 different Atomfast. how will i be able to select one with a Fenix 6? (mac address...?)
I know that iOS Atom app can crash in the bus or train where there are dozens of bluetooth devices offering handshake. Can you test your device in a place where there are 50 people and see if BTE passive check crashes the app?

**Comment by newenclave** [comment]
1) You can select a device in "Scan". Unfortunately Garmin doesn't allow to get mac addresses of devices, so there are should be one of them laying close to the Garmin device. The closest atomfast should have the maximum rssi value. 

2) For now I'm far away from any place with buses or trains :)

**Comment by mickrose555** [comment]
> 1) You can select a device in "Scan". Unfortunately Garmin doesn't allow to get mac addresses of devices, so there are should be one of them laying close to the Garmin device. The closest atomfast should have the maximum rssi value. 
> 
how do you filter atomfast from other BTE devices? during scanning
> 
> 2) For now I'm far away from any place with buses or trains :)

**Comment by newenclave** [comment]
> how do you filter atomfast from other BTE devices? during scanning

By checking of supported BLE services

**Comment by newenclave** [comment]
@mickrose555  Ok. Now I have several answers here. 
Yes. It's possible to connect several AtomFasts to a Garmin device.  My application doesn't support it yet, but I'm going to think about this architecture and UX (yeah I'm really bad in UX).
I dont know how many devices can be connected at the same time. The Garmin documentation says nothing about this and the Garmin forum says it depends on the device/SDK version. Unfortunately I have the only AtomFast and can't check it. 
Today I went to a supermarket with my atom device and tried to find it there. According to my Android there were about 18-20 devices around. No problems where found with this. I don't think there are problems with this. But anything can happen. It needs to be tested on different Garmins.

**Comment by mickrose555** [comment]
> @mickrose555  Ok. Now I have several answers here. 
> 
> Yes. It's possible to connect several AtomFasts to a Garmin device.  My application doesn't support it yet, but I'm going to think about this architecture and UX (yeah I'm really bad in UX).
> 
> I dont know how many devices can be connected at the same time. The Garmin documentation says nothing about this and the Garmin forum says it depends on the device/SDK version. Unfortunately I have the only AtomFast and can't check it. 
> 
> Today I went to a supermarket with my atom device and tried to find it there. According to my Android there were about 18-20 devices around. No problems where found with this. I don't think there are problems with this. But anything can happen. It needs to be tested on different Garmins. 
> 
> 

great!
now remember there are 2 ways, as confirmed on any iOS with Atom freeApp: real pairing (active) and passive reading of multiple atomfast. i am sharing these details just to inform you, not to request this nice-to-have feature. when opening iOS atom app, it scans BLE, filters only atomfast (this is where it can crash in a public place->bug) and reads cps value for each listed/detected atomfast. at this point, no device is selected. nice feature also: each device gets a nickname which is remembered. useful for me because the atomfast left part and right part of the car help me understand where is the anomaly...
(i really like the Fenix 6 sapphire so I will try to get it-> i would never have it if it wasn't for your app!!!)
thanks
